signer/signer_test.go

```diff
@@ -203,3 +204,37 @@ func TestNoSuitableKeyFound(t *testing.T) {
 		t.Fatalf("expected to fail with no suitable key found but failed with: %v", err)
 	}
 }
+
+func TestGetRandReaderNoHSM(t *testing.T) {
+	tcfg := new(Configuration)
+	tcfg.isHsmAvailable = false
+	randReader := tcfg.GetRandReader()
+	if randReader == nil {
+		t.Fatal("expected io.Reader but got nil")
+	}
+	switch randReader.(type) {
+	case io.Reader:
+		break
+	default:
+		t.Fatalf("expected io.Reader but got %T", randReader)
+	}
+}
+
+func TestGetRandReaderHSM(t *testing.T) {
+	tcfg := new(Configuration)
+	tcfg.HSMIsAvailable()
+	randReader := tcfg.GetRandReader()
+	if randReader == nil {
+		t.Fatal("expected io.Reader but got nil")
+	}
+	switch randReader.(type) {
+	case io.Reader:
+		break
+	default:
+		t.Fatalf("expected io.Reader but got %T", randReader)
+	}
+	numBytesRead, err := randReader.Read([]byte("000"))
```

Comment:
What are we testing here? Does this differentiate between the readers because the HSM reader errors on this operation? This [works](https://play.golang.org/p/fCJ9U3EoUwr) for the `crypto/rand` `Reader` case.